packages/components/src/select-control/index.js

```diff
@@ -25,10 +25,15 @@ const initialState = { isExpanded: false, isFocused: false, query: '' };
 export class SelectControl extends Component {
 	constructor( props ) {
 		super( props );
+
+		const { selected, options } = props;
 		this.state = {
 			...initialState,
 			searchOptions: [],
-			selectedIndex: 0,
+			selectedIndex:
+				selected && options?.length
+					? options.findIndex( ( option ) => option.key === selected )
+					: 0,
```

Comment:
What do you think about defaulting to no selection (`-1`)?

That way, when a user tabs over the control without interacting, nothing is selected unexpectedly.

Reply:
Yep makes sense, changes made in 168c4c4.